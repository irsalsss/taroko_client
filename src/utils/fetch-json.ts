import { env } from "@/config/env";

import * as uuid from "uuid";

const BASE_URL = typeof window === "undefined" ? env.API_URL : "";

export const fetchJson = async <JSONDataType = unknown>(
  input: RequestInfo,
  init?: RequestInit
): Promise<JSONDataType> => {
  const url = BASE_URL + input;

  const response = await fetch(url, {
    ...init,
    headers: {
      ...init?.headers,
      "Content-Type": "application/json",
      "Accept": "application/json",
      "X-Request-ID": uuid.v4(),
    },
  });

  const data = await response.json();

  if (!response.ok) {
    throw new Error('Something went wrong')
  }

  return data;
};

export default fetchJson;
