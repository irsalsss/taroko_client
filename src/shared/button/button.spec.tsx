import Button from "./button";
import { render } from '@/utils/test'

describe("Button", () => {
  it("should render primary button successfully", () => {
    const { baseElement } = render(<Button />)

    expect(baseElement).toMatchSnapshot();
    expect(baseElement).toBeTruthy();
  });
});
