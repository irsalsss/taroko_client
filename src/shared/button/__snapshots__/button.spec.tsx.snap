// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render primary button successfully 1`] = `
<body>
  <div>
    <div>
      Button
    </div>
  </div>
</body>
`;
